### PR TITLE
test: Object data type should not change from `beforeSave` to `afterSave` trigger

### LINKED ISSUE #9176: Object data type change from beforeSave to afterSave hook trigger
### New Issue Checklist

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Before posting search [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
Object field data type changes from json to array during beforeSave and afterSave hook triggers.

### Steps to reproduce
https://github.com/parse-community/parse-server/pull/9177

### Actual Outcome
data type of the field value is not same in beforeSave and afterSave hook. It changes from json object to array.

### Expected Outcome
data type of the field value should remain same in beforeSave and afterSave hook.

### Environment
ParseServer: 7.1.0 (7.0.0 no issue observed)

Server
- Parse Server version: `7.1.0`
- Operating system: `Ubuntu 20.04`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Docker`

### PR DESCRIPTION
**Pull request by mkmandar123**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue
Change in data of the object from beforeSave to afterSave hook.

Closes: #9176

## Tasks

- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

**Comment by mtrezza** [comment]
After the recent fixes in Parse Server, let's rebase and see if this test still fails...

File: spec/ParseObject.spec.js [hunk 1]
```diff
@@ -2120,6 +2120,20 @@ describe('Parse.Object testing', () => {
     await object.save();
   });
 
+  it('should not change the json field to array in afterSave', async () => {
```
Comment (mtrezza) [inline spec/ParseObject.spec.js:2123]:
```suggestion
  fit('should not change the json field to array in afterSave', async () => {
```

File: spec/ParseObject.spec.js [hunk 2]
```diff
@@ -2120,6 +2120,20 @@ describe('Parse.Object testing', () => {
     await object.save();
   });
 
+  fit('should not change the json field to array in afterSave', async () => {
```
Comment (mtrezza) [inline spec/ParseObject.spec.js]:
```suggestion
  it('should not change the json field to array in afterSave', async () => {
```

**Comment by mtrezza** [comment]
Test passes; merging this PR to add the additional test to avoid regression and closing related issue.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.2.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.2.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.2.0](https://github.com/parse-community/parse-server/releases/tag/7.2.0)

**Pull request merged**